pkgs/by-name/da/daggerfall-unity/package.nix

```diff
@@ -0,0 +1,110 @@
+{
+  alsa-lib,
+  autoPatchelfHook,
+  copyDesktopItems,
+  fetchzip,
+  lib,
+  libGL,
+  libXScrnSaver,
+  libXcursor,
+  libXi,
+  libXinerama,
+  libXrandr,
+  libXxf86vm,
+  libpulseaudio,
+  libudev0-shim,
+  makeDesktopItem,
+  nix-update-script,
+  stdenv,
+  vulkan-loader,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "daggerfall-unity";
+  version = "1.1.1";
+
+  src = fetchzip {
+    url = "https://github.com/Interkarma/daggerfall-unity/releases/download/v${finalAttrs.version}/dfu_linux_64bit-v${finalAttrs.version}.zip";
+    stripRoot = false;
+    hash = "sha256-JuhhVLpREM9e9UtlDttvFUhHWpH7Sh79OEo1OM4ggKA=";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  buildInputs = [
+    alsa-lib
+    copyDesktopItems
+    libGL
+    libXScrnSaver
+    libXcursor
+    libXi
+    libXinerama
+    libXrandr
+    libXxf86vm
+    libpulseaudio
+    libudev0-shim
+    vulkan-loader
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir --parents "$out"
+    cp --recursive * "$out"
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    patchelf \
```

Comment:
Does `autoPatchelfHook` not handle automatically fixing this? What about executing `autoPatchelf` directly?

Reply:
this may work instead of `postFixup`
```nix
  appendRunpaths = map (pkg: "${lib.getLib pkg}/lib")
    ([ stdenv.cc.libc stdenv.cc.cc ] ++ finalAttrs.buildInputs);
```

Reply:
Either or could be fine, though I'm more inclined to try using `autoPatchelf` first, since it's designed for that.

My main concern is that having such a long and manual `patchelf` is prone to breakage in future updates, and having a more automated approach would definitely be better.

Reply:
appendRunpaths is using autopatchelf

Reply:
I've found two other packages referencing Unity, and both use `patchelf`:

- https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/clonehero/default.nix
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ya/yarg/package.nix

When using `autoPatchelf` the build succeeds but the runtime hangs at this point:

```console
❯ ~/nixpkgs/result/DaggerfallUnity.x86_64 
Mono path[0] = '/nix/store/c9xzmqidbp72ph54vf3i9rzhbapd6sv4-daggerfall-unity-1.1.1/DaggerfallUnity_Data/Managed'
Mono config path = '/nix/store/c9xzmqidbp72ph54vf3i9rzhbapd6sv4-daggerfall-unity-1.1.1/DaggerfallUnity_Data/MonoBleedingEdge/etc'
Preloaded 'lib_burst_generated.so'
Desktop is 0 x 0 @ 0 Hz
```

Reply:
> this may work instead of `postFixup`
> 
> ```nix
>   appendRunpaths = map (pkg: "${lib.getLib pkg}/lib")
>     ([ stdenv.cc.libc stdenv.cc.cc ] ++ finalAttrs.buildInputs);
> ```

This worked, although I didn't need `cc`.